src/features/winterStorageAreaView/utils.ts

```diff
@@ -55,6 +55,68 @@ export const getIndividualWinterStorageArea = (
   };
 };
 
-export const getCustomers = (data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined): WinterStorageCustomer[] => {
-  return [{ id: 'placehold', name: 'placeholder' }];
+export const getWinterStoragePlaces = (data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined): WinterStoragePlace[] => {
+  if (!data?.winterStorageArea?.properties) {
+    return [];
+  }
+  return data?.winterStorageArea?.properties?.sections.edges
+    .reduce<SectionProperties[]>((acc, sectionEdge) => {
+      if (!sectionEdge?.node?.properties) {
+        return acc;
+      }
+      return [...acc, sectionEdge.node.properties];
+    }, [])
+    .map<WinterStoragePlace[]>(
+      (sectionProperties) =>
+        sectionProperties.places.edges.reduce<WinterStoragePlace[]>((acc, placeEdge) => {
+          if (!placeEdge?.node) {
+            return acc;
+          }
+          const leases = placeEdge?.node?.leases?.edges.reduce<Lease[]>((acc, leaseEdge) => {
+            if (!leaseEdge?.node?.application?.customer) {
+              return acc;
+            }
+            const { status, startDate, endDate } = leaseEdge.node;
+            return [
+              ...acc,
+              {
+                customer: {
+                  id: leaseEdge.node.application.customer.id,
+                  firstName: leaseEdge.node.application.customer.firstName,
+                  lastName: leaseEdge.node.application.customer.lastName,
+                },
+                status,
+                startDate,
+                endDate,
+                isActive: false, // FIXME: Retrieve this from the backend once implemented
+              },
+            ];
+          }, []);
+          return [
+            ...acc,
+            {
+              id: placeEdge.node.id,
+              identifier: sectionProperties.identifier,
+              number: placeEdge.node.number,
+              width: placeEdge.node.width,
+              length: placeEdge.node.length,
+              isActive: placeEdge.node.isActive,
+              leases,
+            },
+          ];
+        }, []) ?? []
+    )
+    .reduce<WinterStoragePlace[]>((acc, arr) => [...acc, ...arr], []);
```

Comment:
Chaining `reduce`, `map` and `reduce` seems a bit inefficient, couldn't the preceding logic be packaged in the first `recude`'s callback?

Reply:
With the refactoring done in the latest changes, is this comment still relevant? Mixing the list concatenation logic to the previous callbacks would IMO make the code harder to read.

Reply:
Looks better!